satpy/readers/seviri_l2_bufr.py

```diff
@@ -104,6 +105,10 @@ def __init__(self, filename, filename_info, filetype_info, with_area_definition=
             self.mpef_header['RectificationLongitude'] = f'E{int(rectification_longitude * 10):04d}'
 
         self.with_adef = with_area_definition
+        if filetype_info['file_type'] == 'seviri_l2_bufr_amv':
+            logging.info("AMV BUFR data cannot be loaded with an area definition. Setting self.with_def = False.")
```

Comment:
This should probably check if `self.with_adef` isn't already False right (ex. `if self.with_adef and filetype_info...:`)? Otherwise users will *always* get this no matter what they do.

Also, the `.info` should maybe be a warning? Or maybe a regular `warnings.warn` instead of a logging message? What do you think?

Reply:
Hi David,

Sorry for the late reply. The idea is to make sure that `self.with_adef` is **always** set to False for the `seviri_l2_bufr_amv` products, regardless of whether an area definition is (in this case wrongly) provided or not. But you are probably right in that checking if `self.with_adef` is _already_ False is probably clearer. I will see to that.

Reply:
As for the `logger.info`, I agree. I will change it to `logger.warning`. Thanks for the suggestion.

Reply:
> The idea is to make sure that self.with_adef is always set to False for the seviri_l2_bufr_amv products

That's fine, but if I'm already following that desired behavior I shouldn't get extra information about how I'm *not* following the rules.

Reply:
I fully agree.  :)